File: Backend/CustomerManagementSystem/src/main/java/com/cms/entity/Customer.java
package com.cms.entity;

import com.fasterxml.jackson.annotation.JsonProperty;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Customer {
	@Id
	@JsonProperty("uuid")
	private String uuid;
	
	@JsonProperty("first_name")
	private String firstName;
	
	@JsonProperty("last_name")
    private String lastName;
    
    @JsonProperty("street")
    private String street;
    
    @JsonProperty("address")
    private String address;
    
    @JsonProperty("city")
    private String city;
    
    @JsonProperty("state")
    private String state;
    
    @JsonProperty("email")
    private String email;
    
    @JsonProperty("phone")
    private String phone;


}

File: frontend/scripts/customer.js
document.getElementById('customerForm').addEventListener('submit', async function(event) {
    event.preventDefault();

    // Retrieve form data
    const customerData = {
        first_name: document.getElementById('firstName').value,
        last_name: document.getElementById('lastName').value,
        street: document.getElementById('street').value,
        address: document.getElementById('address').value,
        city: document.getElementById('city').value,
        state: document.getElementById('state').value,
        email: document.getElementById('email').value,
        phone: document.getElementById('phone').value,
    };

    const messageElement = document.getElementById('message');

    try {
        // Retrieve the JWT token from localStorage
        const token = localStorage.getItem('jwt');

        const response = await fetch('https://hiteshsunbase-production.up.railway.app/customers', {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json',
                'Authorization': `Bearer ${token}` // Include Bearer token for authentication
            },
            body: JSON.stringify(customerData)
        });

        if (response.ok) {
            messageElement.textContent = `Customer created successfully!`;
            messageElement.style.color = 'green';

            // Clear form fields after successful creation
            document.getElementById('customerForm').reset();
        } else {
            messageElement.textContent = 'Failed to create customer. Please try again.';
            messageElement.style.color = 'red';
        }
    } catch (error) {
        console.error('Error:', error);
        messageElement.textContent = 'An error occurred. Please try again later.';
        messageElement.style.color = 'red';
    }
});
